File: Python/sieve_of_eratosthenes.py
def SieveOfEratosthenes(n):

    prime = [True for  i in range(n+1)]
    p = 2
    while p*p <= n:
        
        if prime[p] == True:

            for i in range(p*p, n+1, p):
                prime[i] = False
        
        p += 1

    for p in range(2, n+1):
        if prime[p]:
            print(p, end = " ")

# Driver's Code
if __name__ == '__main__':
    n = 300
    print("Following are the Prime numbers smaller than or equal to n")
    SieveOfEratosthenes(n)

File: SieveofEratosthenes.java
import java.util.*;

public class Solution {
    public static void sieveOfEratosthenes(int n)
    {
        
        boolean prime[] = new boolean[n + 1];
        for (int i = 0; i <= n; i++)
            prime[i] = true;
 
        for (int p = 2; p * p <= n; p++)
        {
            
            if (prime[p] == true)
            {
               
                for (int i = p * p; i <= n; i += p)
                    prime[i] = false;
            }
        }
 
       
        for (int i = 2; i <= n; i++)
        {
            if (prime[i] == true)
                System.out.print(i + " ");
        }
    }
 
   
    public static void main(String args[])
    {
        int n = 30;
        System.out.println("Following are the prime numbers smaller than or equal to " + n);
        sieveOfEratosthenes(n);
    }
}